From 26a6b7d3d16e41d8a5b04abbb6151b3f1438dd79 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 21 Mar 2017 17:41:48 +0800
Subject: [PATCH 1653/5242] MLK-14286-7 ARM: clk: imx7ulp: mpll can only be
 used by mipi dsi

commit  b680caf2cf7bdd36c48cdf1eb854681423adb932 from
https://source.codeaurora.org/external/imx/linux-imx.git

The mpll can only be configured by mipi dphy and it is disabled
by default. And it is also not a fixed frequency clock source.
So replace it with a dummy clock for 'periph_slow_sels' to avoid
any peripheral except mipi to use it for clock source.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit a7c6555e0d79491b939b84af99969b21264ec9f8)

Conflicts:
	arch/arm/mach-imx/clk-imx7ulp.c
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx7ulp.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/imx/clk-imx7ulp.c b/drivers/clk/imx/clk-imx7ulp.c
index 87ad820..61ff9ad 100644
--- a/drivers/clk/imx/clk-imx7ulp.c
+++ b/drivers/clk/imx/clk-imx7ulp.c
@@ -31,7 +31,7 @@
 static const char *nic_sels[]		= { "firc", "ddr_div", };
 static const char *periph_plat_sels[]	= { "dummy", "nic1_bus", "nic1_div", "ddr_div", "apll_pfd2", "apll_pfd1", "apll_pfd0", "upll", };
 /* the dummy in only a space holder of spll_bus clk */
-static const char *periph_slow_sels[]	= { "dummy", "osc", "mpll", "firc", "ckil", "nic1_bus", "nic1_div", "dummy", };
+static const char *periph_slow_sels[]	= { "dummy", "osc", "dummy", "firc", "ckil", "nic1_bus", "nic1_div", "dummy", };
 static struct clk *clks[IMX7ULP_CLK_END];
 static struct clk_onecell_data clk_data;
 
-- 
1.7.9.5

